tests/ops/qutrit/test_qutrit_non_parametric_ops.py

```diff
@@ -0,0 +1,122 @@
+from tests.ops.qubit.test_non_parametric_ops import NON_PARAMETRIZED_OPERATIONS
+
+from gate_data import TSHIFT, TCLOCK
+
+NON_PARAMETRIZED_OPERATIONS = [
+    (qml.TShift, TSHIFT),
+    (qml.TClock, TCLOCK),
+]
+
+
+# TODO: Add tests for testing that the decomposition of non-parametric ops is correct
+
+
+class TestOperations:
+    @pytest.mark.parametrize("op_cls, mat", NON_PARAMETRIZED_OPERATIONS)
+    def test_nonparametrized_op_copy(self, op_cls, mat, tol):
+        """Tests that copied nonparametrized ops function as expected"""
+        op = op_cls(wires=range(op_cls.num_wires))
+        copied_op = copy.copy(op)
+        np.testing.assert_allclose(op.matrix(), copied_op.matrix(), atol=tol)
+
+        op._inverse = True
+        copied_op2 = copy.copy(op)
+        np.testing.assert_allclose(op.matrix(), copied_op2.matrix(), atol=tol)
+
+    @pytest.mark.parametrize("ops, mat", NON_PARAMETRIZED_OPERATIONS)
+    def test_matrices(self, ops, mat, tol):
+        """Test matrices of non-parametrized operations are correct"""
+        op = ops(wires=range(ops.num_wires))
+        res_static = op.compute_matrix()
+        res_dynamic = op.matrix()
+        assert np.allclose(res_static, mat, atol=tol, rtol=0)
+        assert np.allclose(res_dynamic, mat, atol=tol, rtol=0)
+
+
+class TestEigenval:
+    def test_tshift_eigenval(self):
+        """Tests that the TShift eigenvalue matches the numpy eigenvalues of the TShift matrix"""
+        op = qml.TShift(wires=0)
+        exp = np.linalg.eigvals(op.matrix())
+        res = op.eigvals()
+        assert np.allclose(res, exp)
+
+    def test_tclock_eigenval(self):
+        """Tests that the TClock eigenvalue matches the numpy eigenvalues of the TClock matrix"""
+        op = qml.TClock(wires=0)
+        exp = np.linalg.eigvals(op.matrix())
+        res = op.eigvals()
+        assert np.allclose(res, exp)
+
+
+period_three_ops = (
+    qml.TShift(wires=0),
+    qml.TClock(wires=0),
+)
+
+
+class TestPowMethod:
+    @pytest.mark.parametrize("op", period_three_ops)
+    @pytest.mark.parametrize("n", (-5, -2, 1, 4, 7))
+    def test_period_three_1_mod_3(self, op, n):
+        """Tests that ops raised to an integer power == 1 mod 3 are the same as the original op"""
+        assert op.pow(n)[0].__class__ is op.__class__
+
+    @pytest.mark.parametrize("op", period_three_ops)
+    @pytest.mark.parametrize("n", (-4, -1, 2, 5, 8))
+    def test_period_three_2_mod_3(self, op, n):
+        """Tests that ops raised to an integer power == 2 mod 3 are the adjoint of the original op"""
+        op_pow = op.pow(n)[0]
+        assert op_pow.__class__ is op.__class__
+        assert np.allclose(op.matrix().conj().T, op_pow.matrix())
+        assert op_pow.inverse == True
+
+    @pytest.mark.parametrize("op", period_three_ops)
+    @pytest.mark.parametrize("n", (-6, -3, 0, 3, 6))
+    def test_period_three_0_mod_3(self, op, n):
+        """Tests that ops raised to an integer power == 0 mod 3 are empty lists"""
```

Comment:
I'm a bit confused by this test, if you have an op w/period 3 shouldn't it return the identity?

Reply:
I was following the same way that the `pow` method is overridden in `pennylane/ops/qubit/non_parametric_ops.py`. I think either way it makes sense, as we are giving the user a "correct" list of operations regardless of whether it's an empty list or a list with `qml.Identity`. Although, I will note that since `qml.Identity` is not optimized for qutrits, it's probably better to return an empty list anyway.